### PR TITLE
Feature/logging test screens updates onboarding

### PR DESCRIPTION
**Pull request by bionara**
# Description

WIP on onboarding modals - still needs logic to not show on every view (ie from startup info)

NOTE this is being merged into https://github.com/zoe/covid-tracker-react-native/pull/1496 (not develop)
Backend PR / branch for local testing: https://github.com/zoe/covid-app/pull/535

Also adds swipe functionality to modal

Things to note/consider:
* progress bars operate same as all other progress bars in app, and do not sync to gif progress; design is also one solid bar, not 4 separate. 
* Gifs to be optimised - higher quality required (original were ~4x file size, so will find the best compromise)

Figma:
https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/COVID-Symptom-Study-(App)---MASTER?node-id=12017%3A157720

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

image
## Checklist (for submission)
image
image

image

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

https://user-images.githubusercontent.com/3264113/127479615-0efe3f66-a7be-49e8-8482-417406a8b683.mov

- [x] Checked against Figma designs (if relevant) - by ie, not px as they are way off.
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

see top of PR. Put in description to make sure people see them.

### CODE REVIEW

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  },
+  title: {
+    marginBottom: 16,
+    marginTop: 16,
+  },
+  newDesign: {
+    backgroundColor: colors.lightBlueBrand,
+    textAlign: 'center',
+    paddingVertical: 3,
+    paddingHorizontal: 12,
```
Comment (juliatan) [inline src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:91]: Should we use `grid` in themes for various numbers?

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  return (
+    <Modal visible onRequestClose={props.onRequestClose} footerChildren={footerChildren}>
+      <View style={styles.wrapper}>
+        <ProgressBlock>
+          <ProgressStatus maxSteps={4} step={onboardingModalScreenIndex + 1} color={colors.lightBlueBrand} />
+        </ProgressBlock>
+
+        <View style={styles.newDesign}>
+          <RegularText style={styles.newDesignText}>{i18n.t(`covid-test-modal.new-design`)}</RegularText>
```
Comment (juliatan) [inline src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:58]: The designs uses the Medium font. This is an example of what I mentioned previous about the mismatch in names. 😞

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+const styles = StyleSheet.create({
+  wrapper: {
+    padding: 8,
+    alignItems: 'center',
+  },
+  button: {
+    paddingHorizontal: 16,
+    marginTop: 64,
```
Comment (juliatan) [inline src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:77]: There's extra gap here between the GIF and OK button I think. I know Aike was fixing this yesterday, so not sure when you pulled the sticky button modal PR, but you might not need this anymore.

**Review by juliatan — DISMISSED** [review]
Looks great! 

When I tried it out, I felt that it was not always easy to see what had happened upon next click because the new GIF and copy just immediately changed without animation. Don't know what others think...

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx (hunk 2 above)
Comment (bionara) [reply src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:58]: Yep. let's leave as-is in this PR, and address the typography on friday's cleanup dev sesh! cheers for the attention to detail!

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx (hunk 1 above)
Comment (bionara) [reply src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:91]: good plan

File: src/features/covid-tests/CovidTestListScreen.tsx [hunk 4]
```diff
@@ -76,11 +83,17 @@ export default function CovidTestListScreen(props: IProps) {
     ? getTestType(props.route.params.mechanism, props.route.params.is_rapid_test) || 'Lateral'
     : 'Lateral';
 
+  const renderModal = () =>
+    showOnboardingModal ? (
+      <CovidTestListOnboardingModal onRequestClose={() => setShowOnboardingModal(false)} visible={true} />
```
[the remark this answers is not in the document]
Comment (bionara) [reply src/features/covid-tests/CovidTestListScreen.tsx:91]: yep can do! I personally always prefer to pull them out into renderABC() functions to help isolate code for readability

**Comment by bionara** [comment]
> Looks great!
> 
> When I tried it out, I felt that it was not always easy to see what had happened upon next click because the new GIF and copy just immediately changed without animation. Don't know what others think...

I agree. I am going to add some gesture support for back/forward to our modals after lunch, and will see if I can add in another option for "dismiss transition" while i am at it! Thanks Julia!!

File: src/components/Modal.tsx [hunk 5]
```diff
@@ -27,30 +32,75 @@ const INSETS = {
...
+
+  function onSwipeLeft() {
+    props.swipeLeft ? props.swipeLeft() : null;
+  }
+
+  function onSwipeRight() {
+    props.swipeRight ? props.swipeRight() : null;
+  }
+
+  function hasSwipe(): boolean {
```
Comment (aikewoody) [inline src/components/Modal.tsx:51]: What is the advantage of making a function if it's only used once?

File: src/components/Modal.tsx [hunk 6]
```diff
@@ -27,30 +32,75 @@ const INSETS = {
...
+      </RNModal>
+    );
+  }
+
+  return hasSwipe() ? (
+    <GestureRecognizer
+      onSwipeUp={(state) => onSwipeUp(state)}
+      onSwipeDown={(state) => onSwipeDown(state)}
+      onSwipeLeft={(state) => onSwipeLeft(state)}
+      onSwipeRight={(state) => onSwipeRight(state)}
```
Comment (aikewoody) [inline src/components/Modal.tsx:98]: To make the code shorter, what about:
`onSwipeUp={(state) => props.swipeUp && props.swipeUp(state)}`

And then you can remove the part:
```
  function onSwipeUp() {
    props.swipeUp ? props.swipeUp() : null;
  }
```

File: src/components/Modal.tsx (hunk 5 above)
Comment (bionara) [reply src/components/Modal.tsx:59]: Tidies up the code where the check happens

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx (hunk 3 above)
Comment (bionara) [reply src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:107]: removed!

**Review by selvach — APPROVED** [review]

**Comment by selvach** [comment]
Wow... tests as well :)

**Pull request merged**